NMS-13426 improve readability of docs for SNMP Config REST (#3522)
OpenNMS/opennms: docs/modules/development/pages/rest/snmp_configuration.adoc

- *Version 1:* If the output only has attributes `community`, `port`, `retries`, `timeout`, and `version`.
- *Version 2:* If there are more attributes than described before (e.g., max Repetitions).

== API version 1

Version 1, supports only a few attributes defined in `snmp-config.xsd`.

> These are defined in `snmp-info.xsd`:

Author: I'm going to need to get another opinion on this, as I'm not sure what the technically correct answer is.

> [options="header"]

Reviewer: In line 62, there is a missing word: 

All parameters can be set regardless of the version.

.Example 1:
[source, bash]
----
curl -v -u admin:admin -X PUT -H "Content-Type: application/xml" \
     -H "Accept: application/xml" \
     -d "<snmp-info>
             <community>yRuSonoZ</community>
             <port>161</port>
             <retries>1</retries>
             <timeout>2000</timeout>
             <version>v2c</version>
          </snmp-info>" \
     http://localhost:8980/opennms/rest/snmpConfig/10.1.1.1
----

> Creates or updates a `<definition/>`-entry for IP address 10.1.1.1 in `snmp-config.xml`.

Reviewer: ```suggestion
Creates or updates a `<definition/>` entry for IP address 10.1.1.1 in `snmp-config.xml`.
```

.Example 2:
[source, bash]
----
curl -v -u admin:admin -X GET ://localhost:8980/opennms/rest/snmpConfig/10.1.1.1
----

Returns the SNMP configuration for IP address 10.1.1.1 as defined in example 1.

== API version 2

> Since version 2, all attributes of a `<definition />` entry defined in `snmp-config.xsd` (http://xmlns.opennms.org/xsd/config/snmp) can be set or get via the interface - except it is only possible to set the configuration for one IP address and not for a range of IP addresses.

Reviewer: ```suggestion
Since version 2, you can set or get all attributes of a `<definition />` entry defined in `snmp-config.xsd` (http://xmlns.opennms.org/xsd/config/snmp) via the interface, except it is only possible to set the configuration for one IP address and not for a range of IP addresses.
```

Author: ```suggestion
Since version 2, all attributes of a `<definition />` entry defined in `snmp-config.xsd` (http://xmlns.opennms.org/xsd/config/snmp) can be set or get via the interface, however it is only possible to set the configuration for one IP address and not for a range of IP addresses.
```

This may change in the future.

The interface uses `SnmpInfo` objects for communication.

> Therefore it is possible to set, for example, v1 and v3 parameters in one request (e.g., `readCommunity` String and `privProtocol` String).

Reviewer: ```suggestion
Therefore it is possible to set, for example, v1 and v3 parameters in one request (e.g., `readCommunity` string and `privProtocol` string).
```
In general, I don't think string needs to be capitalized unless it's part of the syntax.

Author: ```suggestion
Therefore it is possible to set, for example, v1 and v3 parameters in one request (for example, `readCommunity` string and `privProtocol` string).
```

> However {page-component-title} does not allow this.

Reviewer: ```suggestion
However, {page-component-title} does not allow this.
```

> It lets you set only attributes that have no version restriction (for example, timeout value) or the attributes that are limited to the version (for example, `readCommunity` String if version is v1/v2c).

Reviewer: ```suggestion
It lets you set only attributes that have no version restriction (for example, timeout value) or the attributes that are limited to the version (for example, `readCommunity` string if version is v1/v2c).
```

> [options="header, %autowidth"]

Reviewer: I am unable to quote lines 153 and 154. Suggest the following:

The following table shows all supported attributes and the mapping between `snmp-info.xsd` and `snmp-config.xsd`.
It also shows the version limitations, default values, and the restrictions, if any.

| readCommunity           | read-community            | public       | v1, v2c               |
| writeCommunity          | write-community           | private      | v1, v2c               |
| securityName            | security-name             | opennmsUser  | v3                    |
| securityLevel           | security-level            | noAuthNoPriv | v3                    | Integer value, which can be `null`, `1`, `2`, or `3`.

> See note below table for explanation of these values.

Reviewer: ```suggestion
See note below table for an explanation of these values.
```

| authPassPhrase          | auth-passphrase           | 0p3nNMSv3    | v3                    |

> | authProtocol            | auth-protocol             | MD5          | v3                    | only `MD5`, `SHA`, `SHA-224`, `SHA-256`, `SHA-512` are valid arguments

Reviewer: ```suggestion
| authProtocol            | auth-protocol             | MD5          | v3                    | Only `MD5`, `SHA`, `SHA-224`, `SHA-256`, `SHA-512` are valid arguments.
```

| privPassPhrase          | privacy-passphrase        | 0p3nNMSv3    | v3                    |

> | privProtocol            | privacy-protocol          | DES          | v3                    | only `DES`, `AES`, `AES192`, or `AES256` are valid arguments.

Reviewer: ```suggestion
| privProtocol            | privacy-protocol          | DES          | v3                    | Only `DES`, `AES`, `AES192`, or `AES256` are valid arguments.
```

| engineId                | engine-id                 |              | v3                    |
| contextEngineId         | context-engine-id         |              | v3                    |
| contextName             | context-name              |              | v3                    |
| enterpriseId            | enterprise-id             |              | v3                    |
|===

For the `securityLevel` field:

* 1 means `noAuthNoPriv`.
* 2 means `authNoPriv`.
* 3 means `authPriv`.

If you do not set the security level manually it is determined automatically:

> * If no `authPassPhrase` set, the securityLevel is 1.

Reviewer: ```suggestion
* If no `authPassPhrase` set, the security level is 1.
```

Reviewer: I was not clear if this was an actual field or the concept of security level. Whichever one it is, it should be consistent in this file and the snmp_configuration.adoc file.

Author: should be one word to reference the implied field value.

* If `authPassPhrase` is set but no `privPassPhrase` is set, the security level is 2.
* If both a `authPassPhrase` and a `privPassPhrase` are set, the security level is 3.

.Example 1:
[source, bash]
----
curl -v -u admin:admin -X PUT -H "Content-Type: application/xml" \
     -H "Accept: application/xml" \
     -d "<snmp-info>
             <readCommunity>yRuSonoZ</readCommunity>
             <port>161</port>
             <retries>1</retries>
             <timeout>2000</timeout>
             <version>v2c</version>
          </snmp-info>" \
     http://localhost:8980/opennms/rest/snmpConfig/10.1.1.1
----

> Creates or updates a `<definition/>`-entry for IP address 10.1.1.1 in `snmp-config.xml`.

Reviewer: ```suggestion
Creates or updates a `<definition/>` entry for IP address 10.1.1.1 in `snmp-config.xml`.
```
